components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,220 @@
+package softwareversionservice
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/getosutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/pkg/errors"
+)
+
+type ISoftwareVersionService interface {
+	GetSoftwareVersionDetails(string) (*models.SoftwareVersionDetails, error)
+}
+
+type SoftwareVersionService struct {
+	cmdCheckArray  []string
+	osFilePath     string
+	kernelFilePath string
+	logger         logger.Logger
+	CheckPath      func(cmd string) (string, error)
+}
+
+func NewSoftwareVersionService(logger logger.Logger, CheckPath func(cmd string) (string, error)) ISoftwareVersionService {
```

Comment:
NewSoftwareVersionService should return an object of type SoftwareVersionService but not the interface